### PR TITLE
Queries and updates on vet clinic database.

### PR DESCRIPTION
**Pull request by simplegoose**
- Queries are done inside transactions.
- Queries containing aggregate functions.
- Queries to insert a new set of data.

### Screenshots
Screenshot from 2023-05-02 20-13-13
Screenshot from 2023-05-02 20-14-22
Screenshot from 2023-05-02 20-16-42

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -12,3 +12,5 @@ CREATE TABLE animals (
     NEUTERED BOOLEAN, 
     WEIGHT_KG DECIMAL
 );
+
+ALTER TABLE animals ADD SPECIES TEXT;
```
Comment (bobb-Rob) [inline schema.sql]: - [ ] [OPTIONAL] You did well with this query. But I see that used a data type `text` instead of its counter party `varchar(255)`. Below is a brief comparison between them. In SQL, `VARCHAR(255)` and `TEXT` are both data types used to store character string data. However, there are some differences between them:

- Storage size: VARCHAR(255) has a fixed storage size that is equal to the length of the characters it's storing, up to the maximum length of 255 characters. On the other hand, TEXT has a variable storage size and can store up to 2 GB of text.

- Retrieval performance: Retrieving data from a VARCHAR(255) column is generally faster than from a TEXT column because VARCHAR(255) columns are stored in-row, while TEXT columns are stored out-of-row.
- Indexing: VARCHAR(255) columns can be indexed, which can improve lookup performance. TEXT columns can be indexed as well, but with some technical limitations.
- Usage: VARCHAR(255) is commonly used for storing shorter text values such as names, addresses, or emails which are usually shorter than 255 characters. TEXT is used to store large text values like long-form content, descriptions, or comments.

In summary, VARCHAR(255) is ideal for storing relatively short text values while TEXT is better suited for storing large amounts of text data. It is important to choose the appropriate data type based on the nature of the data and its usage requirement as it can affect storage, performance, and retrieval time.

File: queries.sql [hunk 2]
```diff
@@ -8,3 +9,45 @@ SELECT NAME, ESCAPE_ATTEMPTS FROM animals WHERE WEIGHT_KG>10.5;
...
 SELECT * FROM animals WHERE NAME!='Gabumon';
 SELECT * FROM animals WHERE WEIGHT_KG BETWEEN 10.4 AND 17.3;
+
+/* Task Two */
+
+BEGIN;
+
+UPDATE animals SET SPECIES='unspecified';
+
+ROLLBACK;
```
Comment (bobb-Rob) [inline queries.sql]: - [ ] Please ensure you add a query to verify the change before and after rollback as shown below 👇🏾 

```sql
   BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 3]
```diff
@@ -8,3 +9,45 @@ SELECT NAME, ESCAPE_ATTEMPTS FROM animals WHERE WEIGHT_KG>10.5;
...
+
+BEGIN;
+
+UPDATE animals SET SPECIES='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET SPECIES='pokemon' WHERE species IS NULL;
+
+COMMIT;
```
Comment (bobb-Rob) [inline queries.sql]: - [ ] Similarly, as specified in the project requirement, you should provide a query that verifies the change before and after the `commit` as shown below 👇🏾 

```sql  
 BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

File: queries.sql [hunk 4]
```diff
@@ -8,3 +9,45 @@ SELECT NAME, ESCAPE_ATTEMPTS FROM animals WHERE WEIGHT_KG>10.5;
...
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
```
Comment (bobb-Rob) [inline queries.sql]: - [ ] Similarly here, please provide a query that **_verifies_** that the change was effected before and after the `rollback`.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @simplegoose, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql [hunk 5]
```diff
@@ -1,6 +1,16 @@
 /* Populate database with sample data. */
 
+/* Task One */
```
Comment (HAMMAS-SALEEM) [inline data.sql:3]: image

**[OPTIONAL]**

- [ ] It would be nice if you fix these links in your Table of Contents which you can see in the image above to make your readme descriptive and professional 😊

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @simplegoose 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following Requirements
✔️ Implemented Database Perfectly 🥇

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers, and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
